src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -270,13 +309,47 @@ def get(self, request: Request, project: Project) -> Response:
                     )
                     and project.organization.flags.codecov_access
                 )
+
+                # Get Codecov data
                 if should_get_codecov_data:
+                    # Get commit sha from Git blame
+                    ref = (
+                        ctx["commit_id"]
+                        if ctx["commit_id"]
+                        else current_config["config"]["defaultBranch"]
+                    )
+                    should_get_commit_sha = (
+                        len(integrations.filter(provider="github")) > 0
+                        and ref == current_config["config"]["defaultBranch"]
+                    )
+
+                    if should_get_commit_sha:
+                        try:
+                            integration = integrations.filter(provider="github")[0]
+                            integration_installation = integration.get_installation(
+                                organization_id=project.organization_id
+                            )
+                            line_no = ctx.get("line_no")
+                            if line_no:
+                                ref = self.get_commit_sha(
+                                    integration_installation,
+                                    int(line_no),
+                                    filepath,
+                                    current_config["repository"],
+                                    current_config["config"]["defaultBranch"],
+                                )
+                        except Exception:
+                            logger.exception(
+                                "Failed to get commit sha from git blame, pending investigation. Continuing execution."
+                            )
+
                     try:
                         result["lineCoverage"], result["codecovUrl"] = get_codecov_data(
-                            repo=current_config["config"]["repoName"],
```

Comment:
Keep the keys for the beginning of each value. It makes reading the call easier.